src/DatabaseProvider/test.jsx

```
@@ -0,0 +1,53 @@
+import React from 'react'
+import * as TestRenderer from 'react-test-renderer'
+import Database from '../Database'
+import { MockProject, MockTask, MockComment } from '../__tests__/testModels'
+import DatabaseProvider, { Consumer, withDatabase } from '.'
+
+// Simple mock component
+function MockComponent() {
+  return <span />
+}
+
+describe('watermelondb/DatabaseProvider', () => {
+  let database
+  beforeAll(() => {
+    database = new Database({
+      adapter: { schema: null },
+      modelClasses: [MockProject, MockTask, MockComment],
+    })
+  })
+  it('throws if no database or adapter supplied', () => {
+    expect(() => {
+      TestRenderer.create(
+        <DatabaseProvider>
+          <p />
+        </DatabaseProvider>,
+      )
+    }).toThrowError(/You must supply a database/i)
+  })
+  it('passes database to consumer', () => {
+    const instance = TestRenderer.create(
+      <DatabaseProvider database={database}>
+        <Consumer>{db => <MockComponent database={db} />}</Consumer>
+      </DatabaseProvider>,
+    )
+    const component = instance.root.find(MockComponent)
+    expect(component.props.database).toBeDefined()
```

Isn't this redundant? toBeInstanceOf should throw if Database is undefined